### ISSUE TITLE
fix dependencies for embedded tsservice

### ISSUE THREAD

**Issue by bossie** [Open-EO/openeo-geopyspark-driver#356]
Promotion of dev OpenEO (20230221-1118, https://jenkins.vgt.vito.be/job/openEO/job/openeo-integrationtests/1118/) to production had to be rolled back because the embedded tsservice is broken.

Looks like some transitive dependencies changed in a backwards-incompatible way and this manifests itself in e.g. the `/internal/validate` endpoint because this exception shows up in the stderr logs:

```
java.lang.NoClassDefFoundError: be/vito/eodata/gwcgeotrellis/opensearch/OpenSearchClient$
    at be.vito.eodata.extracttimeseries.geotrellis.LayersConfiguration$.be$vito$eodata$extracttimeseries$geotrellis$LayersConfiguration$$s1GrdSigma0ValueLayerConfig(LayersConfiguration.scala:36)
    at be.vito.eodata.extracttimeseries.geotrellis.LayersConfiguration.<init>(LayersConfiguration.scala:313)
    at be.vito.eodata.ts.TimeSeriesValidationService$LayerValidation.<init>(TimeSeriesValidationService.java:56)
    at be.vito.eodata.ts.TimeSeriesValidationService.getValidate(TimeSeriesValidationService.java:106)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
    at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)
    at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:124)
    at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:167)
    at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:219)
    at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:79)
    at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:475)
    at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:397)
    at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81)
    at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:255)
    at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
    at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
    at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
    at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
    at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
    at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
    at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234)
    at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:680)
    at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:394)
    at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:346)
    at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:366)
    at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:319)
    at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:205)
    at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:845)
    at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:583)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
    at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
    at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:226)
    at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1180)
    at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:511)
    at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
    at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1112)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
    at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:213)
    at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:119)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
    at org.eclipse.jetty.server.Server.handle(Server.java:523)
    at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:320)
    at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251)
    at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:273)
    at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:95)
    at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93)
    at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:303)
    at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
    at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136)
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
    at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
    at java.base/java.lang.Thread.run(Thread.java:829)
```

I believe this is the dependency lineage:

ExtractTimeSeries:4.5.0-SNAPSHOT
depends_on
openeo-geotrellis:2.3.0_2.12-SNAPSHOT
depends_on
openeo-opensearch-client:1.0.0_2.12-SNAPSHOT (used to be gwc-geotrellis-core:0.17.0_2.12-SNAPSHOT?)

**Comment by bossie** [comment]
Related? https://github.com/orgs/Open-EO/projects/4/views/2?pane=issue&itemId=21118944

**Comment by bossie** [comment]
Note: https://proba-v-mep.esa.int/api/timeseries/internal/validate is still broken after the rollback so it probably was already broken before the promotion, Icinga seems to confirm this.

TimeSeriesServicesOpenEO:4.5.0-SNAPSHOT is built against openeo-geotrellis:2.3.0_2.12-SNAPSHOT, which in turn used to depend on gwc-geotrellis-core:0.17.0_2.12-SNAPSHOT, but now depends on openeo-opensearch-client:1.0.0_2.12-SNAPSHOT instead.

When geotrellis-extensions/openeo-geotrellis stopped depending on gwc-geotrellis-core, the class above was no longer available for TimeSeriesServicesOpenEO at runtime.

**tl;dr** just get the develop branch of geotrellistimeseries to build again (the LayersConfig classes have already been updated): https://jenkins.vgt.vito.be/job/openEO/job/geotrellistimeseries/job/develop/

**Comment by bossie** [comment]
There's a test that keeps on failing with an OOME. Jenkins could only tell me that it happened in `GeotrellisTimeSeriesSpec` but analysis of the heap dump showed that `"return a result for each polygon in the input collection"` is the culprit:

![test_failed_oome](https://user-images.githubusercontent.com/1032518/222484355-56a40a7c-21c0-438d-a911-b7cd671b6642.png)

**Comment by bossie** [comment]
Fixed more tests and was able to produce new, more stable, 4.5.0-SNAPSHOT artifacts: https://jenkins.vgt.vito.be/job/openEO/job/geotrellistimeseries/job/develop/29/

**Comment by bossie** [comment]
Confirmed:

:heavy_check_mark:  /apidocs/
:heavy_check_mark: /internal/health
:heavy_check_mark: /v1.0/ts
:heavy_check_mark: /v1.0/ts/S2_FAPAR_FILE/geometry?startDate=2017-01-01&endDate=2018-01-01
:x:  /internal/validate:
```json
{
    "errorMessage": "Time Series Validation Error: Mol/2014-01-01: expected 0.68, was NaN"
}

```

**Comment by bossie** [comment]
As discussed, not fixing /internal/validate.
